### PR TITLE
Communication(1): common files

### PR DESCRIPTION
**Pull request by teoperisanu**
Add common files for SPI, I2C, UART.

### CODE REVIEW

File: bindings/csharp/CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,8 @@ INCLUDE_DIRECTORIES(
 	${CMAKE_SOURCE_DIR}/include
 	${CMAKE_CURRENT_SOURCE_DIR})
 
-SET(CMAKE_SWIG_FLAGS "")
+set(CMAKE_SWIG_FLAGS "")
+
```
Comment (AlexandraTrifan) [inline bindings/csharp/CMakeLists.txt]: This can be removed from the changes, right?

File: bindings/python/CMakeLists.txt [hunk 2]
```diff
@@ -12,7 +12,11 @@ INCLUDE_DIRECTORIES(${PYTHON_INCLUDE_PATH}
 	${CMAKE_SOURCE_DIR}/include
 	${CMAKE_CURRENT_SOURCE_DIR})
 
-SET(CMAKE_SWIG_FLAGS "")
+if(ENABLE_TOOLS)
+	set(CMAKE_SWIG_FLAGS -DCOMMUNICATION)
```
Comment (AlexandraTrifan) [inline bindings/python/CMakeLists.txt]: Do we need to have "-DCOMMUNICATION" or just -DCOMMUNICATION  is fine?

Comment (teoperisanu) [reply bindings/python/CMakeLists.txt]: For one flag the quotes are not necessary.

File: bindings/csharp/CMakeLists.txt (hunk 1 above)
Comment (teoperisanu) [reply bindings/csharp/CMakeLists.txt]: Removed in commit d74555b.

**Pull request merged**